scripts/makeNetwork.py

```diff
@@ -147,7 +148,8 @@ def qemuArchNetworkConfig(i, arch, n):
         else:
             vlan_id = vlan if vlan else i
             mac_str = "" if not mac else ",macaddr=%s" % mac
-            return "-net nic,vlan=%(VLAN)i%(MAC)s -net tap,vlan=%(VLAN)i,id=net%(I)i,ifname=${TAPDEV_%(I)i},script=no" % { 'I' : i, 'MAC' : mac_str, 'VLAN' : vlan_id}
+            #workaround for qemu >= 2.1.1 tested on Ubuntu 19 with qemu 3.1
+            return "-netdev tap,id=net%(I)i,ifname=${TAPDEV_%(I)i},script=no -device e1000,netdev=net%(I)i" % { 'I' : i, 'MAC' : mac_str, 'VLAN' : vlan_id}
```

Comment:
You can drop the `VLAN` field as well, but you need to append the MAC address field to the end of the `device` argument, like `netdev=net%(I)i%(MAC)s`.

Reply:
Fixed  in the commit `bf57905`.